### ISSUE TITLE
AnalyzerTelemetry should not be a public type

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#5974]
This API should be make internal.

**Comment by mavasani** [comment]
/cc @srivatsn @mattwar 

We added this type based on the feedback from a design meeting when we designing replacing the IDE analyzer driver with compilation event based driver. IDE needs to report telemetry data for analyzer actions being executed, and this information is not publically available. We decided that we should create a telemetry specific public type in the compiler layer and add the API to fetch analyzer action counts to report in telemetry.

**Comment by mavasani** [comment]
Based on our internal discussions:
We need **some** means to access analyzer action counts from the core compiler layer for reporting telemetry in IDE –analyzer execution is now a blackbox for the v2 IDE analyzer driver. Our alternatives are:

1. Retain the current public API 
2. Propose a better/refined API 
3. Remove the API and don’t report analyzer action counts as part of telemetry

I’d prefer (1), but am open to (2) if there is a proposal.

**Comment by mavasani** [comment]
@DustinCampbell recommended the below API shape, I'll make changes accordingly:

```C#
public class AnalyzerTelemetryInfo
{
    TimeSpan ExecutionTime { get; }

    int CompilationStartActionsCount { get; }
    int CompilationEndActionsCount { get; }
    int CompilationActionsCount { get; }
    int SyntaxTreeActionsCount { get; }
    int SemanticModelActionsCount { get; }
    int SymbolActionsCount { get; }
    int SyntaxNodeActionsCount { get; }
    int CodeBlockStartActionsCount { get; }
    int CodeBlockEndActionsCount { get; }
    int CodeBlockActionsCount { get; }
}

public class CompilationWithAnalyzers
{
    public AnalyzerTelemetryInfo GetAnalyzerTelemetryInfoAsync(DiagnosticAnalyzer analyzer, CancellationToken cancellationToken);
}

```

**Comment by mavasani** [comment]
I am going to address this API change as part of https://github.com/dotnet/roslyn/issues/6044
